lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1416,13 +1416,32 @@ synthesizeObservedSetterBody(AccessorDecl *Set, TargetImpl target,
   SmallVector<ASTNode, 6> SetterBody;
 
   auto callObserver = [&](AccessorDecl *observer, VarDecl *arg) {
+    // If we're not loading the old value, then re-write the observer's type to
+    // be either () -> () or (Self) -> () -> () and remove the oldValue
+    // parameter.
+    if (!arg) {
+      observer->setParameters(ParameterList::createEmpty(Ctx));
+      auto emptyTy = TupleType::getEmpty(Ctx);
+      auto param = FunctionType::Param(emptyTy);
+      if (SelfDecl) {
+        param = FunctionType::Param(SelfDecl->getType());
```

Comment:
This is the wrong way around, you’re building () -> (Self) -> () instead of (Self) -> () -> (). Also this won’t work if the type is generic. But ideally you shouldn’t need this code at all; calling computeType() on the observer should do the trick.

Reply:
Oops, `computeType` is a really nice simplification. Thanks!